app/controllers/orders_controller.rb

```diff
@@ -1,16 +1,26 @@
 class OrdersController < ApplicationController
   def show
-    if !current_user || current_user != (@order = Order.find(params[:id])).user
-      @access_denied = true
+    if @order = Order.find_by(id: params[:id])
```

Comment:
Make use of `before_action`, cancan's `authorize_resource`, and maybe `rescue_from` (some people dismiss the use of `rescue_from` for redirection but my opinion is that Rails does that too so why can't we?) in order to simplify controller actions.

The rule of thumb is that you want to standardize redirection in a way that is simple and easy to predict. Don't customize too much.